synapse/lib/stormlib/oauth.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+
+import yarl
+from oauthlib import oauth1
+
+import synapse.exc as s_exc
+import synapse.lib.stormtypes as s_stormtypes
+
+@s_stormtypes.registry.registerLib
+class OAuthV1Lib(s_stormtypes.Lib):
+    '''
+    A Storm library to handle oauth v1 authentication
+    '''
+    _storm_locals = (
+        {
+            'name': 'client',
+            'desc': '''
+                Initialize an OAuthV1 Client to use for signing/authentication.
+            ''',
+            'type': {
+                'type': 'function', '_funcname': '_methClient',
+                'args': (
+                    {'name': 'ckey', 'type': 'str',
+                     'desc': 'The OAuthV1 Consumer Key to store and use for sigining requests.'},
+                    {'name': 'csecret', 'type': 'str',
+                     'desc': 'The OAuthV1 Consumer Secret used to sign requests.'},
+                    {'name': 'atoken', 'type': 'str',
+                     'desc': 'The OAuthV1 Access Token (or resource owner key) to use to sign requests.)'},
+                    {'name': 'asecret', 'type': 'str',
+                     'desc': 'The OAuthV1 Access Token Secret (or resource owner secret) to use to sign requests.'},
+                    {'name': 'sigtype', 'type': 'str', 'default': oauth1.SIGNATURE_TYPE_QUERY,
+                     'desc': 'Where to populate the signature (in the HTTP body, in the query parameters, or in the header)'},
+                ),
+                'returns': {
+                    'type': 'storm:oauth:v1:client',
+                    'desc': 'An OAuthV1 client to be used to sign requests.',
+                }
+            },
+        },
+    )
+
+    _storm_lib_path = ('inet', 'oauth', 'v1',)
```

Comment:
haha, hate to thrash around, but i think this belongs within `inet.http`.  Could we rename to `inet.http.oauth.v1`?